<launch>
  <!-- RL Training Mode Launch File -->
  <!-- This launches Gazebo in headless mode for faster RL training -->
  
  <arg name="headless" default="true"/>
  <arg name="real_time" default="false"/>
  <arg name="paused" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="gui" default="false"/>
  <arg name="world_name" default="worlds/empty.world"/>
  
  <!-- Load robot description -->
  <param name="robot_description" command="$(find xacro)/xacro $(find New_description)/urdf/New.xacro"/>
  
  <!-- Start Gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world_name)"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="headless" value="$(arg headless)"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="physics" default="ode"/>
    <arg name="verbose" default="false"/>
    <arg name="extra_gazebo_args" default=""/>
  </include>
  
  <!-- Spawn robot in Gazebo -->
  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" 
        args="-param robot_description -urdf -model New -z 0.1" 
        output="screen"/>
  
  <!-- Load controller configuration -->
  <rosparam file="$(find New_description)/launch/controller.yaml" command="load"/>
  
  <!-- Start controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" 
        respawn="false" output="screen" ns="New" 
        args="Joint_1_position_controller Joint_2_position_controller Joint_3_position_controller Joint_4_position_controller joint_state_controller"/>
  
  <!-- Robot state publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" 
        respawn="false" output="screen">
    <remap from="/joint_states" to="/New/joint_states"/>
  </node>
  
  <!-- Optional: Add target visualization marker -->
  <node name="target_marker" pkg="New_description" type="target_marker.py" output="screen" if="$(arg gui)"/>
  
</launch>